pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
 
 __all__ = [
     'CHTest',
+    'decompose',
     'OCSBTest'
 ]
 
 
+def decompose(x, type, f, filter=None):
+    """
+
+    Parameters
+    ----------
+    x : np.array
+        The time series of which the trend, seasonal, and noise/random
+        components will be extracted.
+
+    type: str
+        The type of decomposition that will be performed - 'multiplicative' or
+        'additive'. We would use 'multiplicative' generally when we see an
+        increasing trend. We use 'additive' when the trend is relatively
+        stable over time.
+
+    f: int
```

Comment:
Can we rename this `m`? We use `m` across the rest of the package to denote seasonal frequency.

Reply:
So I used `f` because that is what was used internally within the R source code for the `decompose` function. I think we can put `m` and just put a little note that it sometimes denoted as `f` within the R code.

Reply:
Yeah, but we're not 1:1 with R. I'd rather be consistent within our package than across different software.

Reply:
Done!